File: solutions/typescript/resistor-color/1/resistor-color.ts
export const COLORS: string[] = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
];

export const colorCode: (color:string) => number|Error = (color: string) => {
  const i: number = COLORS.indexOf(color);
  return (i > -1) ? i : new Error("Not supported Color");
}

File: solutions/go/resistor-color/1/resistor_color.go
package resistorcolor

import "slices"

// Colors returns the list of all colors.
func Colors() (s []string) {
    s = append(s, "black")
    s = append(s, "brown")
    s = append(s, "red")
    s = append(s, "orange")
    s = append(s, "yellow")
    s = append(s, "green")
    s = append(s, "blue")
    s = append(s, "violet")
    s = append(s, "grey")
    s = append(s, "white")
	return 
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	return slices.Index(Colors(), color)
}
